src/3d/qgsrubberband3d.h

```diff
@@ -116,49 +131,119 @@ class _3D_EXPORT QgsRubberBand3D
      */
     MarkerType markerType() const;
 
+    /**
+     * Sets the marker outline style
+     * \note Default value is solid line
+     * \since QGIS 3.44
+     */
+    void setMarkerOutlineStyle( Qt::PenStyle style );
+
+    /**
+     * Returns current marker outline style used by rubberband.
+     * \since QGIS 3.44
+     */
+    Qt::PenStyle markerOutlineStyle() const;
+
+    /**
+     * Sets whether to show markers on vertices
+     * \since QGIS 3.44
+     */
+    void setMarkerEnabled( bool enable );
+
+    /**
+     * Returns whether markers are being shown on vertices
+     * \since QGIS 3.44
+     */
+    bool markerEnabled() const;
+
+    /**
+     * Sets whether to show edges
+     * \since QGIS 3.44
+     */
+    void setEdgesEnabled( bool enable );
+
+    /**
+       * Returns whether edges are shown
+       * \since QGIS 3.44
+       */
+    bool edgesEnabled() const;
+
+    /**
+       * Sets whether to show polygon fill
+       * \since QGIS 3.44
+       */
+    void setPolygonFillEnabled( bool enable );
+
+    /**
+       * Returns whether polygon fill is shown
+       * \since QGIS 3.44
+       */
+    bool polygonFillEnabled() const;
+
     void reset();
 
     void addPoint( const QgsPoint &pt );
 
-    void setPoints( const QgsLineString &points );
+    /**
+     * Sets the geometry of rubberband
+     *
+     * \note Function doesn't set \a GeometryType
```

Comment:
I think setGeometry() should set the geometry type as well, otherwise it may be quite surprising, if I set polygon geometry, but the geometry type would stay line/point. Also QgsRubberBand::setToGeometry() updates the geometry type...